staging/src/k8s.io/client-go/util/certificate/certificate_manager.go

```diff
@@ -174,8 +184,21 @@ type manager struct {
 	clientFn         CSRClientFunc
 	stopCh           chan struct{}
 	stopped          bool
+
+	// wraps time.Now but can be stubbed out for testing
+	clock clock
```

Comment:
can this be just a func field?
`now func() time.Time`

Reply:
Oh yeah great idea. Done.